app/parse.py

```diff
@@ -15,9 +42,76 @@ class Product:
     num_of_reviews: int
 
 
-def get_all_products() -> None:
-    pass
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def get_single_product(product: WebElement) -> Product:
+    return Product(
+        title=product.find_element(By.CSS_SELECTOR, "h4 > a").get_attribute(
+            "title"
+        ),
+        description=product.find_element(By.CLASS_NAME, "description").text,
+        price=float(
+            product.find_element(By.CLASS_NAME, "price").text.replace("$", "")
+        ),
+        rating=len(product.find_elements(By.CLASS_NAME, "ws-icon-star")),
+        num_of_reviews=int(
+            product.find_element(By.CLASS_NAME, "review-count").text.split()[0]
+        ),
+    )
+
+
+def get_all_products(products: [WebElement]) -> [Product]:
+    return [get_single_product(product) for product in products]
+
+
+def write_products_to_csv_file(
+        products: [Product], output_csv_path: str
+) -> None:
+    with open(output_csv_path, "w", encoding="utf-8", newline="") as file:
+        writer = csv.writer(file)
+        writer.writerow(PRODUCT_FIELDS)
+        writer.writerows([astuple(product) for product in products])
+
+
+def click_more_button(driver) -> None:
+    while True:
+        try:
+            more_button = driver.find_element(
+                By.CLASS_NAME, "ecomerce-items-scroll-more"
+            )
+        except NoSuchElementException:
+            break
+        else:
+            if not more_button.is_displayed():
+                break
+            more_button.click()
+            time.sleep(0.1)
```

Comment:
https://selenium-python.readthedocs.io/waits.html#explicit-waits
try to use WebDriverWait